asciidoctorj-core/src/main/java/org/asciidoctor/jruby/cli/AsciidoctorInvoker.java

```diff
@@ -148,34 +147,31 @@ private URLClassLoader createUrlClassLoader(List<String> classPaths) {
 
     private String renderInput(Asciidoctor asciidoctor, Options options, List<File> inputFiles) {
 
-        if(inputFiles.size() == 1 && "-".equals(inputFiles.get(0).getName())) {
+        if (inputFiles.size() == 1 && "-".equals(inputFiles.get(0).getName())) {
             return asciidoctor.convert(readInputFromStdIn(), options);
         }
 
-        StringBuilder output = new StringBuilder();
-
-        for (File inputFile : inputFiles) {
-
-            if (inputFile.canRead()) {
-
-                String renderedFile = asciidoctor
-                        .convertFile(inputFile, options);
-                if (renderedFile != null) {
-                    output.append(renderedFile).append(
-                            System.getProperty("line.separator"));
-                }
-            } else {
-                System.err.println("asciidoctor: FAILED: input file(s) '"
-                        + inputFile.getAbsolutePath()
-                        + "' missing or cannot be read");
-                throw new IllegalArgumentException(
-                        "asciidoctor: FAILED: input file(s) '"
-                                + inputFile.getAbsolutePath()
-                                + "' missing or cannot be read");
-            }
-        }
+        findInvalidInputFile(inputFiles)
+                .ifPresent(inputFile -> {
+                    System.err.println("asciidoctor: FAILED: input file(s) '"
+                            + inputFile.getAbsolutePath()
+                            + "' missing or cannot be read");
+                    throw new IllegalArgumentException(
+                            "asciidoctor: FAILED: input file(s) '"
+                                    + inputFile.getAbsolutePath()
+                                    + "' missing or cannot be read");
+                });
+
+        return inputFiles.stream()
```

Comment:
I refactored this but the output is actually never used, maybe we should remove and just cal convert "for each" input?

Reply:
Yes, that would make sense. The method is also not part of any public API, so we can just remove the return value.